src/main/java/com/thebois/models/beings/Pawn.java

```diff
@@ -38,4 +49,29 @@ private void setRandomDestination() {
         setDestination(randomPosition);
     }
 
+    protected void pickNearestIncompleteStructureAsDestination() {
+        final Collection<Optional<IStructure>> structures =
+            getFinder().findNearestStructures(this.getPosition(), 10);
+        final Boolean[] found = {false};
```

Comment:
Might be missing something obvious but why the array? Seems like we only set the first element to either true or false?